### PR TITLE
Pass profile explicitly to CDK CLI when using `npx ampx sandbox --profile <profile_name>`

### PR DESCRIPTION
**Pull request by sobolk**
## Problem

Passing profile via environment variables (see [here](https://github.com/aws-amplify/amplify-backend/blob/b2464ab051a1e4b8380fda8ebd911a610d64ac6e/packages/cli/src/command_middleware.ts#L35-L37)) is not enough to assure consistency.

Customer using `npx ampx sandbox --profile <profile_name>` might run into edge case scenarios where local or user home scoped cdk configuration might override environemnt.
See [code in CDK](https://github.com/aws/aws-cdk/blob/c7980a2e068b5336914364050f72e41259aebc90/packages/aws-cdk/lib/settings.ts#L134-L139). I.e. config is resolved in such a way that env variables have the least priority (implicitly).

**Issue number, if available:**

## Changes

Pass profile explicitly when calling CDK commands. So that it has the same priority as in ampx cli.

**Corresponding docs PR, if applicable:**

## Validation

1. Added tests.
2. Manual validations with debugger.

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/sandbox/API.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ export type Sandbox = {
 // @public (undocumented)
 export type SandboxDeleteOptions = {
     identifier?: string;
+    profile?: string;
```
Comment (sobolk) [inline packages/sandbox/API.md:28]: surprisingly `SandboxOptions` below already have `profile` but was unused.

Comment (rtpascual) [reply packages/sandbox/API.md:28]: Looks like on our docs both `sandbox` and `sandbox delete` are mentioned to have `profile` which leads to confusion

**Review by rtpascual — APPROVED** [review]

File: packages/sandbox/API.md (hunk 1 above)
Comment (sobolk) [reply packages/sandbox/API.md:28]: Which is fine, they both are supposed to handle that (except edge cases I described in PR description).
I was just surprised that `SandboxOptions.profile` already exists but noting in the code was reading (using) it to do anything useful.

File: packages/cli/src/commands/sandbox/sandbox_command.test.ts [hunk 2]
```diff
@@ -205,6 +205,46 @@ void describe('sandbox command', () => {
...
+    // User said yes to delete
+    contextual.mock.method(AmplifyPrompter, 'yesOrNo', () =>
+      Promise.resolve(true)
+    );
+
+    const profile = 'test_profile';
+    await commandRunner.runCommand(`sandbox --profile ${profile}`);
+
+    // Similar to the later 0ms timeout. Without this tests in github action are failing
+    // but working locally
```
Comment (awsluja) [inline packages/cli/src/commands/sandbox/sandbox_command.test.ts:232]: is it possible something is missing in the code? I don't see this required in line [360](https://github.com/aws-amplify/amplify-backend/pull/1900/files#diff-fe3062a0618a08be0e78b23acc94e8621cd2d865ad1b63da87b19278b8fe06c7R360)

Comment (awsluja) [reply packages/cli/src/commands/sandbox/sandbox_command.test.ts:232]: possibly a CI flag in the yargs tool?

Comment (sobolk) [reply packages/cli/src/commands/sandbox/sandbox_command.test.ts:232]: The other test you mentioned is not testing SIGINT.

I took this from test above. I.e. line 197.

I wanted to cover a change where profile is passed to sandbox delete that happens when you say "yes" after CTRL+C.

Comment (awsluja) [reply packages/cli/src/commands/sandbox/sandbox_command.test.ts:232]: do we need to re-import yargs in our beforeEach? https://github.com/yargs/yargs/blob/c36c571e4e15dfe26be1d919e4991fb6ab6ed9fd/test/yargs.js#L20

Comment (sobolk) [reply packages/cli/src/commands/sandbox/sandbox_command.test.ts:232]: Afaik we can't re-import yargs while using ESM imports. 

This solution was not invented in this PR.
See https://github.com/aws-amplify/amplify-backend/pull/100#discussion_r1280702279

**Review by awsluja — APPROVED** [review]

**Pull request merged**